pytensor/scalar/basic.py

```diff
@@ -1140,14 +1140,23 @@ def output_types(self, types):
         else:
             raise NotImplementedError(f"Cannot calculate the output types for {self}")
 
+    @staticmethod
+    def _cast_scalar(x, dtype):
+        if hasattr(x, "astype"):
+            return x.astype(dtype)
+        else:
+            return x
```

Comment:
Does something like this work?
```suggestion
            return getattr(np, dtype)(x)
```

Reply:
interesting